### PR TITLE
[Storage] [Typing] [Blob] `_container_client.py`

### PR DESCRIPTION
**Pull request by vincenttran-msft**

### CODE REVIEW

File: sdk/storage/azure-storage-blob/azure/storage/blob/_container_client.py [hunk 1]
```diff
@@ -1174,34 +1170,34 @@ def delete_blob(
 
     @overload
     def download_blob(
```
Comment (vincenttran-msft) [inline sdk/storage/azure-storage-blob/azure/storage/blob/_container_client.py:1172]: azure\storage\blob\_container_client.py:1172: error: Overloaded function signatures 1 and 2 overlap with incompatible return types  [misc]

Same as in blob_client

File: sdk/storage/azure-storage-blob/azure/storage/blob/_container_client.py [hunk 2]
```diff
@@ -762,7 +758,7 @@ def set_container_access_policy(
         access_conditions = get_access_conditions(lease)
         timeout = kwargs.pop('timeout', None)
         try:
-            return self._client.container.set_access_policy(
+            return self._client.container.set_access_policy(  #type: ignore
```
Comment (vincenttran-msft) [inline sdk/storage/azure-storage-blob/azure/storage/blob/_container_client.py:761]: Another case of seems like generated code has typehints wrong

File: sdk/storage/azure-storage-blob/azure/storage/blob/_container_client.py [hunk 3]
```diff
@@ -1578,7 +1574,7 @@ def get_blob_client(
                 DeprecationWarning
             )
         try:
-            blob_name = blob.get('name')
+            blob_name = blob.get('name')  # type: ignore
```
Comment (vincenttran-msft) [inline sdk/storage/azure-storage-blob/azure/storage/blob/_container_client.py:1577]: We will likely have to keep this to maintain the typehint as `str` to dissaude people from using the "deprecated" `BlobProperties`

File: sdk/storage/azure-storage-blob/azure/storage/blob/_container_client.py [hunk 4]
```diff
@@ -1295,7 +1295,7 @@ def download_blob(
             **kwargs)
 
     @distributed_trace
-    def delete_blobs(
+    def delete_blobs(  # pylint: disable=delete-operation-wrong-return-type
```
Comment (vincenttran-msft) [inline sdk/storage/azure-storage-blob/azure/storage/blob/_container_client.py:1298]: `azure/storage/blob/_container_client.py:1298: [C4752(delete-operation-wrong-return-type), ContainerClient.delete_blobs] delete* or begin_delete* should return None or LROPoller[None], respectively.`

**Review by jalauzon-msft — APPROVED** [review]

**Pull request merged**